app/controllers/course_user_data_controller.rb

```diff
@@ -106,7 +106,8 @@ def update
         if @editCUD.save
           redirect_to(action: :show) && return
         else
-          redirect_to(action: :edit) && return
+          @editCUD.tweak ||= Tweak.new
+          render :edit
```

Comment:
For the record, I don't really like that this doesn't redirect on a POST request, but for the time being, I think it's the best solution we have.